### PR TITLE
feat(v2 volume/configurable cpu mask): updata crds.yaml and longhorn.yaml

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#8835

### CODE REVIEW

**Review by innobead — CHANGES_REQUESTED** [review]
Don't we need to add the new setting `v2-data-engine-cpu-mask` to the chart?

**Comment by derekbit** [comment]
> Don't we need to add the new setting v2-data-engine-cpu-mask to the chart?

Yes, I forgot it. Added, thank you.

**Comment by innobead** [comment]
> > Don't we need to add the new setting v2-data-engine-cpu-mask to the chart?
> 
> Yes, I forgot it. Added, thank you.

Let's remember to go through the items in https://github.com/longhorn/longhorn/wiki/Longhorn-Setting-Update, as we need to update the Chart readme and questions.yaml. It's a bit lengthy, but we can potentially streamline it using automation later.. 

@longhorn/doc needs to review the description of the new setting as usual.

File: chart/values.yaml [hunk 1]
```diff
@@ -323,6 +323,8 @@ defaultSettings:
   v2DataEngineHugepageLimit: ~
   # -- Number of millicpus on each node to be reserved for each Instance Manager pod when the V2 Data Engine is enabled. The default value is "1250".
   v2DataEngineGuaranteedInstanceManagerCPU: ~
+  # --The setting is used to specify which CPU cores the Storage Performance Development Kit (SPDK) target daemon within each instance manager pod should run on. The default value is "0x1".
```
Comment (jillian-maroket) [inline chart/values.yaml:326]:
```suggestion
  # --CPU cores on which the Storage Performance Development Kit (SPDK) target daemon should run. The SPDK target daemon is located in each Instance Manager pod. The default value is "0x1".
```

**Review by jillian-maroket — COMMENTED** [review]
I split up the first sentence in the description to make it easier to read.

File: chart/values.yaml (hunk 1 above)
Comment (derekbit) [reply chart/values.yaml]: @jillian-maroket Can you review it again? Thank you.

File: chart/values.yaml [hunk 2]
```diff
@@ -323,6 +323,8 @@ defaultSettings:
   v2DataEngineHugepageLimit: ~
   # -- Number of millicpus on each node to be reserved for each Instance Manager pod when the V2 Data Engine is enabled. The default value is "1250".
   v2DataEngineGuaranteedInstanceManagerCPU: ~
+  # -- CPU cores on which the Storage Performance Development Kit (SPDK) target daemon should run. The SPDK target daemon is located in each Instance Manager pod. The default value is "0x1". The number of cores should be equal to or smaller than the guaranteed instance manager CPU for v2 data engine.
```
Comment (jillian-maroket) [inline chart/values.yaml:326]:
```suggestion
  # -- CPU cores on which the Storage Performance Development Kit (SPDK) target daemon should run. The SPDK target daemon is located in each Instance Manager pod. Ensure that the number of cores is less than or equal to the guaranteed Instance Manager CPUs for the V2 Data Engine. The default value is "0x1".
```

**Review by jillian-maroket — COMMENTED** [review]
I made a few changes to the new sentence.

**Comment by innobead** [comment]
@derekbit it seems you haven't updated the questions.yaml and chart readme yet?

> Let's remember to go through the items in [Wiki: Longhorn Setting Update](https://github.com/longhorn/longhorn/wiki/Longhorn-Setting-Update), as we need to update the Chart readme and questions.yaml. It's a bit lengthy, but we can potentially streamline it using automation later..

**Comment by derekbit** [comment]
> @derekbit it seems you haven't updated the questions.yaml and chart readme yet?
> 
> > Let's remember to go through the items in [Wiki: Longhorn Setting Update](https://github.com/longhorn/longhorn/wiki/Longhorn-Setting-Update), as we need to update the Chart readme and questions.yaml. It's a bit lengthy, but we can potentially streamline it using automation later..

Oh, yes, forgot it. I probably need a workflow to check it.

**Comment by derekbit** [comment]
cc @jillian-maroket ready for review. Thank you.

File: chart/questions.yaml [hunk 3]
```diff
@@ -850,19 +833,13 @@ questions:
...
-    WARNING:
-      - If the backing image name is not specified, setting this field is meaningless.
-      - As for backing image creation with data source type \"upload\", it is recommended to do it via UI rather than StorageClass here. Uploading requires file data sending to the Longhorn backend after the object creation, which is complicated if you want to handle it manually.'
+    description: 'Data source type of a backing image used in a Longhorn StorageClass. If the backing image exists in the cluster, Longhorn uses this setting to verify the image. If the backing image does not exist, Longhorn creates one using the specified data source type. WARNING: - If the backing image name is not specified, setting this field is meaningless. - As for backing image creation with data source type \"upload\", it is recommended to do it via UI rather than StorageClass here. Uploading requires file data sending to the Longhorn backend after the object creation, which is complicated if you want to handle it manually.'
```
Comment (jillian-maroket) [inline chart/questions.yaml:842]:
```suggestion
    description: 'Data source type of a backing image used in a Longhorn StorageClass. If the backing image exists in the cluster, Longhorn uses this setting to verify the image. If the backing image does not exist, Longhorn creates one using the specified data source type. **Caution** [1] This field is useful only if the backing image name is specified. [2] Backing images with data source type \"upload\" are best created using the Longhorn UI. Uploading requires sending file data to Longhorn after object creation, which can be complicated when performed manually.'
```

File: chart/questions.yaml [hunk 4]
```diff
@@ -873,10 +850,7 @@ questions:
     - "export-from-volume"
     default: ""
   - variable: persistence.backingImage.dataSourceParameters
-    description: "Data source parameters of a backing image used in a Longhorn StorageClass. You can specify a JSON string of a map. (Example: `'{\"url\":\"https://backing-image-example.s3-region.amazonaws.com/test-backing-image\"}'`)
-    WARNING:
-      - If the backing image name is not specified, setting this field is meaningless.
-      - Be careful of the quotes here."
+    description: "Data source parameters of a backing image used in a Longhorn StorageClass. You can specify a JSON string of a map. (Example: `'{\"url\":\"https://backing-image-example.s3-region.amazonaws.com/test-backing-image\"}'`) WARNING: - If the backing image name is not specified, setting this field is meaningless. - Be careful of the quotes here."
```
Comment (jillian-maroket) [inline chart/questions.yaml:853]:
```suggestion
    description: "Data source parameters of a backing image used in a Longhorn StorageClass. You can specify a JSON string of a map. (Example: `'{\"url\":\"https://backing-image-example.s3-region.amazonaws.com/test-backing-image\"}'`) **Caution** [1] This field is useful only if the backing image name is specified. [2] Ensure that quotes are used correctly when specifying parameters."
```

File: chart/questions.yaml [hunk 5]
```diff
@@ -956,8 +930,7 @@ questions:
   type: boolean
   group: "Other Settings"
 - variable: networkPolicies.enabled
-  description: "Setting that allows you to enable network policies that control access to Longhorn pods.
-  Warning: The Rancher Proxy will not work if this feature is enabled and a custom NetworkPolicy must be added."
+  description: "Setting that allows you to enable network policies that control access to Longhorn pods. Warning: The Rancher Proxy will not work if this feature is enabled and a custom NetworkPolicy must be added."
```
Comment (jillian-maroket) [inline chart/questions.yaml:933]:
```suggestion
  description: "Setting that allows you to enable network policies that control access to Longhorn pods. **Caution** The Rancher Proxy will work only if this feature is enabled and a custom NetworkPolicy is added."
```

File: chart/questions.yaml [hunk 6]
```diff
@@ -970,17 +943,18 @@ questions:
...
+    - "rke2"
+    - "k3s"
   - variable: defaultSettings.v2DataEngineGuaranteedInstanceManagerCPU
     label: Guaranteed Instance Manager CPU for V2 Data Engine
-    description: 'Number of millicpus on each node to be reserved for each Instance Manager pod when the V2 Data Engine is enabled. The default value is "1250".
-    WARNING:
-    - Specifying a value of 0 disables CPU requests for instance manager pods. You must specify an integer between 1000 and 8000.
-    - This is a global setting. Modifying the value triggers an automatic restart of the instance manager pods. Do not modify the value while volumes are still attached."
-    group: "Longhorn Default Settings'
+    description: 'Number of millicpus on each node to be reserved for each Instance Manager pod when the V2 Data Engine is enabled. The default value is "1250". WARNING: - Specifying a value of 0 disables CPU requests for instance manager pods. You must specify an integer between 1000 and 8000. - This is a global setting. Modifying the value triggers an automatic restart of the instance manager pods. Do not modify the value while volumes are still attached." group: "Longhorn Default Settings'
```
Comment (jillian-maroket) [inline chart/questions.yaml:951]:
```suggestion
    description: 'Number of millicpu on each node to be reserved for each Instance Manager pod when the V2 Data Engine is enabled. The default value is \"1250\". **Caution** [1] Specifying a value of \"0\" disables CPU requests for Instance Manager pods. You must specify an integer between \"1000\" and \"8000\". [2] This is a global setting. Modifying the value triggers an automatic restart of the Instance Manager pods. Do not modify the value while volumes are still attached.'
```

File: chart/questions.yaml [hunk 7]
```diff
@@ -970,17 +943,18 @@ questions:
...
     type: int
     min: 1000
     max: 8000
     default: 1250
+  - variable: defaultSettings.v2DataEngineCPUMask
+    label: CPU Mask for V2 Data Engine
+    description: 'CPU cores on which the Storage Performance Development Kit (SPDK) target daemon should run. The SPDK target daemon is located in each Instance Manager pod. Ensure that the number of cores is less than or equal to the guaranteed Instance Manager CPUs for the V2 Data Engine. The default value is 0x1.'
```
Comment (jillian-maroket) [inline chart/questions.yaml:958]:
```suggestion
    description: 'CPU cores on which the Storage Performance Development Kit (SPDK) target daemon should run. The SPDK target daemon is located in each Instance Manager pod. Ensure that the number of cores is less than or equal to the guaranteed Instance Manager CPUs for the V2 Data Engine. The default value is \"0x1\".'
```

File: chart/questions.yaml (hunk 6 above)
Comment (jillian-maroket) [reply chart/questions.yaml:951]: > group: "Longhorn Default Settings'

This should not be part of the description.

**Review by jillian-maroket — COMMENTED** [review]
Let me know if you have concerns about the changes.

**Review by jillian-maroket — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**